### PR TITLE
[WIP] Improve release process

### LINKED ISSUE #78: release process & deployment is undocumented
How to do a tagged release with a GitHub release? How do builds end up in use for `market`? Where do builds happen? How is the CHANGELOG.md updated, cause it's empty right now? What are the deployment URLs?

Examples of structure:
- for ocean.js - https://github.com/oceanprotocol/ocean.js#-production
- for ocean.py - https://github.com/oceanprotocol/ocean.py/blob/main/READMEs/release-process.md (but ideally shorter, see [ocean.py#179](https://github.com/oceanprotocol/ocean.py/issues/179))

### PR DESCRIPTION
**Pull request by akshay-ap**
Fixes #78  .

Changes proposed in this PR:

- Add script to generate changelog.
- Changelog is appended to `CHANGELOG.md` file.

### CODE REVIEW

File: release.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+changes="$(git log main..feature/bumpversion-to-v$2 --format=%s --no-merges --invert-grep --grep="Generate changelog for release" | awk '{print "- " $0}')"
+
+touch CHANGELOG.md
+
+echo '#### v'$2$'\n' >> CHANGELOG.md
+echo "> $(date +\"%d-%B-%Y\")" >> CHANGELOG.md
+echo "$changes"$'\n' >> CHANGELOG.md
+
+git add CHANGELOG.md
+git commit -m "Generate changelog for release v"$2
```
Comment (calina-c) [inline release.sh:31]: what happens if the commit fails due to pre-commit checks?

Comment (akshay-ap) [reply release.sh:31]: Thanks for pointing out. I think in that case, changes should be reverted and error message should be printed.

**Review by calina-c — CHANGES_REQUESTED** [review]
waiting for this comment https://github.com/oceanprotocol/provider/pull/117#discussion_r602450660

**Comment by calina-c** [comment]
@akshay-ap Is there any progress on this? Can I help with anything?

**Comment by akshay-ap** [comment]
hi @calina-c, thanks for jumping in. I will resume on this task and will let you know if any issues.

**Comment by calina-c** [comment]
@akshay-ap is this still needed? Was it superseded by the new changes?

**Comment by akshay-ap** [comment]
> 
> 
> @akshay-ap is this still needed? Was it superseded by the new changes?

@calina-c I think this is no longer needed. As agreed with @alexcos20  and @trentmc , we plan to make it easy to follow release steps rather than automate the steps as of now.

**Pull request closed without merging**